controllers/nginx/pkg/cmd/controller/utils.go

```diff
@@ -39,6 +39,18 @@ func sysctlSomaxconn() int {
 	return maxConns
 }
 
+// sysctlFSFileMax returns the value of fs.file-max, i.e.
+// maximum number of open file descriptors
+func sysctlFSFileMax() int {
+	maxConns, err := sysctl.New().GetSysctl("fs/file-max")
+	if err != nil {
+		glog.Errorf("unexpected error reading system maximum number of open file descriptors (fs.file-max): %v", err)
+		return 0
+	}
+
+	return maxConns
```

Comment:
> I honestly just don't know - I'm happy to follow your lead here if you say "this is correct", just not sure I follow the logic of letting a single nginx process use up all the FDs available to the whole system?

Actually this process can use just the available FDs in the system.
If nginx tries to use more than the available the log just will show "Too many open files" which means:

- the current `fs.file-max` value is low
- the node is small
- there is a process using/leaking descriptors

This setting tries to use the available resources (if needed) and not require scale the pod just because we are using the default values and we cannot get more than a couple of thousand connections.